odoo/addons/base/models/res_partner.py

```diff
@@ -234,6 +236,8 @@ def default_get(self, default_fields):
 
     _sql_constraints = [
         ('check_name', "CHECK( (type='contact' AND name IS NOT NULL) or (type!='contact') )", 'Contacts require a name'),
+        ('uniq_company_registry', 'unique(company_registry, country_id, parent_id)',
```

Comment:
This won't work, imagine this use case.

Company A, child B => both company_registry="abcd"
Company C has a child D => D can have company_registry="abcd" too, because it has a different parent_id.

So you would need to ensure that no partner other than one of the parents is using "abcd" which is not done here.

Reply:
Well the child (an individual) cannot modify the `company_registry` so such a case cannot be reproduced I think.

Reply:
You don't need the `parent_id` in the key though. If anything, you should  add a constraint that forbids you to have both a `company_registry` and a `parent_id` (not necessary IMO though)

Reply:
There are some examples in the tests where a child inherits its parent's company registry (through `parent_id` and `category_id`) which then creates a uniqueness problem. This constraint is supposed to fix that by allowing a child to have the same company registry as its parent (even if, as you said, it does not make much sense for an individual to have a company registry per se)

Reply:
> There are some examples in the tests where a child inherits its parent's company registry (through `parent_id` and `category_id`)

Not sure I understand

Reply:
E.g.: in `test_325_50_invoicing_and_paying_subpartner_should_impact_commercial_partner` the `child_partner` has the company registry as its `parent_partner`

Reply:
But it shouldn't since it is a commercial field :thinking:

Reply:
Actually it is normal because:
- We create the parent partner with a VAT
- In Belgium, when the VAT is set, we copy it to the company registry number
- We create the child partner with the `parent_id` set
- `_commercial_sync_from_company` will sync the values (`vat` and `company_registry` as defined in `_commercial_fields`) from the parent to the chield
- Hence, the `child_partner` will have the VAT and Company Registry Number as its parent. The uniqueness constraint is therefore violated.
- So that's why we need to take into account the `parent_id` in the uniqueness constraint